e2e/kaivuilmoitus_muutosilmoitus.spec.ts

```diff
@@ -0,0 +1,469 @@
+    .fill(asianhoitaja.y_tunnus);
+  await page.getByTestId('applicationData.invoicingCustomer.invoicingOperator').click();
+  await page.getByTestId('applicationData.invoicingCustomer.ovt').click();
+  await page.getByTestId('applicationData.invoicingCustomer.ovt').fill('0000000000000');
+  await page.getByTestId('applicationData.invoicingCustomer.invoicingOperator').fill('1');
+  await page.getByRole('button', { name: 'Seuraava' }).click();
+  await page.getByRole('button', { name: 'Seuraava' }).click();
+  await page.getByRole('button', { name: 'Lähetä hakemus' }).click();
+  await page.getByRole('button', { name: 'Vahvista' }).click();
+  await expect(page.getByText('Hakemus lähetetty')).toBeVisible({ timeout: 40000 });
+  await page.getByRole('link').filter({ hasText: /HAI/gm }).click();
+  await expect(page.getByText('Hakemukset', { exact: true })).toBeVisible({ timeout: 10000 });
+  await page.getByText('Hakemukset', { exact: true }).click();
+
+  const kaivuilmoitus = await page
+    .locator('[data-testid^=applicationViewLinkIdentifier-KP]')
+    .textContent();
+  const hanketunnus = await page.locator('[data-testid^=hanke-tunnus]').textContent();
+
+  // käsittely
+  await page.goto(testiData.allu_url);
+  await page.getByPlaceholder('Username').fill(testiData.allupw);
+  await page.getByRole('button', { name: 'Submit' }).click();
+  await expect(page.getByRole('link', { name: 'HAKEMUKSET' })).toBeVisible();
+  await page.getByRole('link', { name: 'HAKEMUKSET' }).click();
+  await expect(page.getByRole('button', { name: 'HAE' })).toBeVisible();
+  await page.getByLabel('Hakemuksen tunnus').fill(`${kaivuilmoitus}`);
+  await page.getByRole('button', { name: 'HAE' }).click();
+  // kaivuilmoitus
+  await expect(page.getByRole('link', { name: `${kaivuilmoitus}` })).toBeVisible({
+    timeout: 20000,
+  });
+  await page.getByRole('link', { name: `${kaivuilmoitus}` }).click();
+  await page.getByRole('button', { name: 'NÄYTÄ UUDET TIEDOT' }).click();
+  await page.getByRole('button', { name: 'KÄSITTELYYN' }).click();
+  await page.getByLabel('Hakemuksen lajit *').getByText('Hakemuksen lajit').click();
+  await page.getByText('Katu- ja vihertyöt').click();
+  await page.locator('.cdk-overlay-container > div:nth-child(3)').click();
+  await expect(page.getByRole('button', { name: 'TALLENNA' })).toBeVisible();
+  await page.getByRole('button', { name: 'TALLENNA' }).click();
+
+  // Odotetaan tuloksia
+  await page.goto(testiData.alluTriggerUrl);
+  await expect(async () => {
+    await page.goto(`${testiData.hankesalkku}${hanketunnus}`);
+    await page.getByText('Hakemukset').click();
+    await expect(page.getByTestId('application-status-tag').first()).toContainText(
+      'Odottaa käsittelyä',
+      {
+        timeout: 2000,
+      },
+    );
+    await expect(page.getByTestId('application-status-tag').nth(1)).toContainText('Käsittelyssä', {
+      timeout: 2000,
+    });
```

Comment:
I don't know if we can assume the applications will always be in the same order, and we only care about the status of the kaivuilmoitus, so we can only check that one.
```suggestion
    await expect(
      page
        .getByTestId('application-card')
        .filter({ hasText: kaivuilmoitus! })
        .getByTestId('application-status-tag'),
    ).toContainText('Käsittelyssä', { timeout: 2000 });
```